### PR TITLE
Toggletip: Add PopoverModal wrapper

### PR DESCRIPTION
**Pull request by adela-almasan**
This PR adds a wrapper for the `Toggletip` to cover the use case when the toggletips acts as a popover. The `theme` prop is set to `popover` in this case.

PopoverModal

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by torkelo — COMMENTED** [review]
worried that very few that use it as a popover will set the right theme, a new PopoverModal component that is just a wrapper around Toggletip with theme="popover" ?

File: packages/grafana-ui/src/components/Toggletip/Toggletip.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { ToggletipContent } from './types';
 
 export interface ToggletipProps {
   /** The theme used to display the toggletip */
-  theme?: 'info' | 'error';
+  theme?: 'info' | 'error' | 'custom';
```
[the remark this answers is not in the document]
Comment (adela-almasan) [reply packages/grafana-ui/src/components/Toggletip/Toggletip.tsx:16]: it does, I'll update.

**Comment by adela-almasan** [comment]
> worried that very few that use it as a popover will set the right theme, a new PopoverModal component that is just a wrapper around Toggletip with theme="popover" ?

@torkelo I think @staton-hyse11 is already [planning that](https://github.com/grafana/grafana/pull/75846#issuecomment-1745361155), I can create a basic wrapper and adapt as the plan settles. Thanks!

**Comment by torkelo** [comment]
great! :+1:

**Review by ashharrison90 — COMMENTED** [review]
this seems a little backwards. `Toggletip` is a specific implementation of a more generic `Popover` component. so let's create a base `Popover` component first, then refactor `Toggletip` to be a thin wrapper around that?

**Comment by ivanortegaalba** [comment]
I agree @ashharrison90. Popover is lower level, and the Toggletip should be built on top. 

So, we need to find a solution to fix the background issue. I see different options here:
- Long term: We create a generic popover component and make Tooltip and Toggletip to use it (High cost, best approach)
- Short term: We add `className` to Toggletip and we modify the styles in the meantime for this use case (Low cost, minimum impact, no deprecation or implications for Toggletip component)
- Short term: Keep the approach of adding the theme, but for me, this one, will have implications when we want to remove it since we have to deprecate it.

Thoughts?

**Comment by ashharrison90** [comment]
@ivanortegaalba where is this new popover going to be used initially? if it's just in core grafana, we can keep this `PopoverModal` wrapper that @adela-almasan has written, put it in `/core/components` for now, (not grafana-ui as that would require maintaining/deprecating it correctly) and just use it whilst we work on option 1 which i think we agree is the option we have to do eventually? 🤔 we can also add a comment or add it to the generic popover issue to remind us to refactor/remove it?

if it needs to be used in external plugins then this wouldn't work obviously 😬

**Comment by adela-almasan** [comment]
@ashharrison90 this is only going to be used for GenAI History  ([here](https://github.com/grafana/grafana/blob/e773f999ff65aaf678d4b619669b311b2e4d4032/public/app/features/dashboard/components/GenAI/GenAIButton.tsx#L144) in code), so it's scope is very narrow for now. I can move it out of `grafana-ui` and it can definitely disappear when we're gonna have a generic popover - this was definitely not intended to be a permanent solution.

**Comment by ashharrison90** [comment]
think the only problem as ivan alluded to is any modifications to `Toggletip` need to be carefully considered, because anything we add we have to support and deprecate properly.

**Comment by adela-almasan** [comment]
@ashharrison90 from what I've seen the `theme` prop is not being used, so everybody is just defaulting to `info`. I'm not sure if adding a new theme will trigger usage, but that's not really measurable at this time. I'm not entirely sure how to move forward with this one.. we could keep the theme and take it into account in case of deprecation or we could allow custom styling via props (imo that could potentially be more disruptive and we could end up with weirdly looking toggletips). What are you more comfortable with? Thoughts?

**Comment by torkelo** [comment]
Alerting also has need of it (the rule editor is using toggletip for  a popover form for query options). 

I think it could be useful from plugins but don't have any concrete examples now. Is toggletip used any from plugins? Maybe check plugins platforms usage database.

**Comment by ashharrison90** [comment]
@adela-almasan think we (frontend platform) need to discuss the best way forward. unfortunately @staton-hyse11 is out for the next week so might need to wait for him to get back 😬 am i right in thinking that nothing is currently blocked, and the only issue we have is that the toggletip background color is slightly wrong for using form elements? if so, i'd rather just wait to have that discussion before we introduce a prop value we probably don't want to support. 

hope that makes sense and sorry for the back and forth 🤦

**Comment by adela-almasan** [comment]
@ashharrison90 yes, nothing is currently blocked, I just wanted to correct the background color. No worries, I know there's a bigger discussion going on with the Popover and some changes might not make sense in that context. Sounds good, let's wait until Staton is back and we'll regroup then. Thanks!

**Comment by nmarrs** [comment]
@ashharrison90  @torkelo we are prepping these features for ObsCon - is there something that you would like addressed / fixed by that point (next two weeks).

Our current understanding is that we will have a new component to address this use case (see [this google doc by Ash](https://docs.google.com/document/d/1O9z8CqrNpS6vz0Iy4Aa5a-fZRIhLzaVSnTVaRsZdEoI/edit#heading=h.wdnhyxwa7rhw))

**Comment by ashharrison90** [comment]
yep I plan to tackle it tomorrow. I don't think there's anything you need to do, only thing that will change is the background color and I don't think anyone will notice tbh

**Pull request closed without merging**